src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/Refresh.cs

```diff
@@ -0,0 +1,122 @@
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.App.ErrorContainers;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Binding.BindInfo;
+using Microsoft.PowerFx.Core.Entities;
+using Microsoft.PowerFx.Core.Errors;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Functions.Publish;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+
+namespace Microsoft.PowerFx.Core.Texl.Builtins
+{
+    // Refresh(source:*[...]) : b
+    // Refresh(source:![...]) : b
+    internal sealed class RefreshFunction : BuiltinFunction
+    {
+        // This is a stateful asynchronous function with side effects.
+        public override bool IsSelfContained => false;
+
+        public override bool ModifiesValues => true;
+
+        public override bool IsStateless => false;
+
+        public override bool IsAsync => true;
+
+        public override bool DisableForDataComponent => true;
+
+        public override Capabilities Capabilities => Capabilities.OutboundInternetAccess | Capabilities.EnterpriseAuthentication | Capabilities.PrivateNetworkAccess;
+
+        public override bool SupportsParamCoercion => false;
+
+        public RefreshFunction()
+            : base("Refresh", TexlStrings.AboutRefresh, FunctionCategories.Table, DType.Boolean, 0, 1, 1)
+        { 
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.RefreshArg1 };
+        }
+
+        // We have suggestions for the first argument.
+        public override bool HasSuggestionsForParam(int argumentIndex)
+        {
+            Contracts.Assert(argumentIndex >= 0);
+
+            return argumentIndex == 0;
+        }
+
+        protected override bool RequiresPagedDataForParamCore(TexlNode[] args, int paramIndex, TexlBinding binding)
+        {
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.Assert(paramIndex >= 0 && paramIndex < args.Length);
+            Contracts.AssertValue(binding);
+            Contracts.Assert(binding.IsPageable(args[paramIndex].VerifyValue()));
+
+            // Refresh only needs metadata. No actual data from datasource is required.
+            return false;
+        }
+
+        public override void CheckSemantics(TexlBinding binding, TexlNode[] args, DType[] argTypes, IErrorContainer errors)
+        {
+            Contracts.AssertValue(binding);
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.AssertAllValid(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+            Contracts.Assert(MinArity <= args.Length && args.Length <= MaxArity);
+
+            if (args[0] is not FirstNameNode)
```

Comment:
>args[0] is not FirstNameNode [](http://example.com/codeflow?start=16&length=28)

This should only be done if binding.Features.PowerFxV1CompatibilityRules is true. Otherwise we should keep the same behavior.